evennia/contrib/slots_handler/slots.py

```diff
@@ -0,0 +1,385 @@
+"""
+Slots Handler
+
+This class is designed to sit on typeclassed objects and allow for other
+objects to be attached to configurable slots on the handler-endowed object.
+
+Built by DamnedScholar (https://github.com/damnedscholar)
+"""
+
+
+from collections import OrderedDict
+import evennia
+from evennia.utils.dbserialize import _SaverDict, _SaverList, _SaverSet
+
+
+class SlotsHandler:
+    """
+    Handler for the slots system. This handler is designed to be attached to
+    objects with a particular @lazy-property name, so it will be referred to as
+    `.slots`, though individual games can set that however they like.
+
+    Place on character or other typeclassed object:
+    ```
+    @lazy_property
+    def slots(self):
+        return SlotsHandler(self)
+    ```
+
+    The purpose of this handler is to sit on a "holder" typeclassed object and
+    manage slots for "held" objects. The first thing that should happen is
+    using `self.slots.add()` on the holder object to identify the available
+    options. `.slots.delete()` can be used to delete or pop old slots.
+
+    By default, a held object can have slots stored on it (the handler will
+    check `.db.slots`, `.ndb.slots`, and `.slots` in that order), and
+    `self.slots.attach()` will attempt to attach it on all of those slots (it
+    should fail if it can't attach on every slot). `self.slots.drop()` will
+    remove the target from any of the slots given it. Both of these methods
+    have the option to be given a custom `slots` argument, which will override
+    the slots on the held object.
+
+    Using the custom `slots` in `attach()` and `drop()` provides some
+    customization facility in that you can store *any* object in a slot, not
+    just an object that has been set up for it.
+    """
+
+    def __init__(self, obj):
+        self.obj = obj
+        self._objid = obj.id
+
+    # Internal utility methods
+    def is_name_valid(self, test):
+        "Internal function to check if the name is right."
+        try:
+            exec(test + " = 1")
```

Comment:
I suggest this entire check should be removed. If this is code only intended for developers, this check is pointless - if a wrong variable name is used a traceback will catch it. 
If this contrib is used by some overlying build command it's much worse - using `exec()` - and completely without checking the input too - you are now opening up yourself to a huge security exploit - if the input `test` (for example passed from some build command to `slots.add or slots.delete`) is in any way in the hands of an untrusted user they can now execute Python code unchecked on your server - the same power as `@py`. Even if you don't intend for more than internal use, the use of `exec()` is not advertised anywhere and may not be caught and the input properly verified by an user of this contrib. So, please remove this.

Reply:
Done.